fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/impl/content/registry/FabricAbstractBlock.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.ai.pathing.PathNodeType;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link AbstractBlock} subclasses.
+ *
+ * <p>Note: This interface is automatically implemented on the {@link AbstractBlock} class
+ * via Mixin and interface injection.
+ *
+ * <p>Note to maintainers: Functions should only be added to this interface if they are general-purpose enough,
+ * to be evaluated on a case-by-case basis. Otherwise, they are better suited for more specialized APIs.
+ */
+public interface FabricAbstractBlock {
+	/**
+	 * Gets the {@link PathNodeType} for the block.
+	 *
+	 * <p>This returns the {@link PathNodeType} if the block is a direct target in an entity path.
+	 *
+	 * <p>For example, in this case, for a cactus-like block you should specify {@link PathNodeType#DAMAGE_CACTUS},
+	 * so entities will avoid to pass through or above the block.
+	 *
+	 * <p>Tip: Return null if you want to use vanilla default behavior.
+	 *
+	 * @param state Current block state.
+	 * @return the {@link PathNodeType} specified for the block.
+	 */
+	@Nullable
+	default PathNodeType getPathNodeType(BlockState state) {
+		return null;
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} for the block if is in a closest neighbor position
+	 * that the entity will find in his path.
+	 *
+	 * <p>This returns the {@link PathNodeType} if the block is a direct target in an entity path.
```

Comment:
I think this is more clear. Also shouldn't this be neighbor here?

```suggestion
	 * Gets the {@link PathNodeType} for the block if it is a direct neighbor position
	 * to an entity path (directly next to a block that the entity will pass through or above).
	 *
	 * <p>This returns the {@link PathNodeType} if the block is a direct neighbor in an entity path.
```